### PR TITLE
Update readme for io buffer redesign.

### PR DESCRIPTION
**Pull request by jyao1**
Fix: https://github.com/DMTF/libspdm/issues/799

Signed-off-by: Jiewen Yao <jiewen.yao@intel.com>

### CODE REVIEW

File: doc/user_guide.md [hunk 1]
```diff
@@ -30,20 +30,41 @@ Please refer to spdm_client_init() in [spdm_requester.c](https://github.com/DMTF
...
+   1.1, allocate buffer for the spdm_context, initialize it, and setup scratch_buffer.
+   The spdm_context may include the decrypted secured message or session key.
+   The scratch buffer may include the decrypted secured message.
+   The spdm_context and scratch buffer shall be zeroed before freed or reused.
 
    ```
    spdm_context = (void *)malloc (libspdm_get_context_size());
    libspdm_init_context (spdm_context);
+
+   scratch_buffer_size = libspdm_get_sizeof_required_scratch_buffer(m_spdm_context);
```
Comment (steven-bellock) [inline doc/user_guide.md:42]: Integrator should also be able to get the size statically to be used in a linker script.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: doc/user_guide.md (hunk 1 above)
Comment (jyao1) [reply doc/user_guide.md:42]: Are you proposing to remove get_scratch_buffer_size, but define a MACRO in spdm_common_lib.h ?

Comment (steven-bellock) [reply doc/user_guide.md:42]: Both ways should be available. It can be called `LIBSPDM_SCRATCH_BUFFER_SIZE` or something similar.

Comment (jyao1) [reply doc/user_guide.md:42]: Doc updated

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**